src/design-pages/passkey_authentication.rst

```diff
@@ -0,0 +1,318 @@
+similar to the “certificate_verification” option for the p11_child, as it
+contains the parameters needed to tune the passkey_child helper
+process. The list of parameters and their meaning:
+
+* user-verification: if set to true, requires user verification (i.e. PIN,
+  password) during authentication. If set to false, does not request user
+  verification during authentication. The default is that the key itself
+  decides what to do.
+
+* debug-libfido2: flag to print libfido2 library messages. Defaults to false.
+
+"passkey_child_timeout" sets the timeout for the PAM responder to wait for
+passkey_child to finish.
+
+Prompting Configuration
++++++++++++++++++++++++
+Another section called [prompting/passkey] is added. This section is similar
+to other prompting sections (i.e. 2fa). The list of options and their meaning:
+
+* interactive: set to prompt a message and wait before testing the presence of
+  a passkey device. Recommended if your device doesn’t have a tactile trigger.
+
+* inter-prompt=your prompt here: set individual prompt message for interactive
+  mode. Default is: “Insert your Passkey device, then press ENTER.”
+
+* touch: set to prompt a message to remind the user to touch the device.
+
+* touch-prompt=your prompt here: set individual prompt message for the cue
+  option. Default is: “Please touch the device."
+
+An example prompt interaction can look like: ::
+
+    Insert your Passkey device, then press ENTER.
+    < ENTER >
+    Enter PIN: 1234
+    Please touch the device.
+
+User verification
++++++++++++++++++
+The user verification can be set in various places:
+
+* In the IPA passkey configuration
+* In the local sssd.conf
+
+IPA passkey configuration user verification requirement overrides local
+sssd.conf.
+
+Registration process
+--------------------
+
+sssctl
+******
+
+Each passkey needs to be registered before it can be used for authentication.
+This registration process is quite simple; the user connects the hardware
+token to the computer, and then, executes the
+``sssctl passkey-exec`` command.
+
+The command contains several parameters that slightly change its behaviour or
+its output. The following is a list of the mandatory options:
+
+* domain: LDAP domain name. Also know as
+  `Relying Party <https://www.w3.org/TR/webauthn-2/#webauthn-relying-party>`__
+  in the WebAuthn standard.
+
+List of optional parameters:
+
+* username: this is the username as registered in the LDAP server.
+
+* type: public key cryptography. There are three possible types: es256, rs256
+  and EdDSA. The default is es256.
+
+* user-verification: requires user verification (i.e. PIN, fingerprint).
+
+* cred-type: credential type (server-side or discoverable).
+
+* debug-libfido2: flag to print libfido2 library messages.
+
+The most basic example of a registration would be the following: ::
+
+    # sssctl passkey-exec --register --username=USERNAME --domain=DOMAIN
```

Comment:
I know I'm late and I haven't reviewed patches that already land in, but I'm not very fond of current implementation.

```
errno_t sssctl_passkey_exec(struct sss_cmdline *cmdline,
                            struct sss_tool_ctx *tool_ctx,
                            void *pvt)
{
    errno_t ret;

    ret = sssctl_wrap_command(SSS_PASSKEY_CHILD, cmdline, tool_ctx, pvt);

    return ret;
}
```

Wouldn't it be better to provide `sssctl passkey-register` command instead of allowing to call anything in passkey_child?

Reply:
That makes sense. @justin-stephenson what do you think?

Reply:
But users need to call only register, right? You can call passkey_child directly for debugging and testing purpose.

Reply:
Opened https://issues.redhat.com/browse/SSSD-5644 to track this work.